src/Core/GlobPatternMatcher.cs

```diff
@@ -64,7 +64,9 @@ private class WildcardPattern
 
             private WildcardPattern(string pattern, string directorySeparator)
             {
-                this.pattern = new Regex(toRegexp(pattern, directorySeparator));
+                this.pattern = new Regex(toRegexp(pattern, directorySeparator),
+                                RegexOptions.Compiled,
+                                RegexConstants.DefaultTimeout);
```

Comment:
The `WildcardPattern` is used every time we analyse a file in Connected Mode to check whether the file is include/excluded by the server settings like `sonar.exclusions`

The patterns don't change very often (only when the server settings are updated) and there are used frequently so I made them `Compiled`.